### PR TITLE
Fix for redis memory check failure after link flap and also sometimes cpu usage high failure

### PR DESCRIPTION
**Pull request by wumiaont**
### Description of PR
Redis memory check result is not a stable value using "redis-cli info memory | grep used_memory_human". It's found on a stable system (BGO converged, no port flapping etc), the above check could have memory usage difference by more than 0.2M. 

Followings are CLI output from 202405 and 202205.

202405:
admin@ixre-egl-board30: redis-cli info memory | grep used_memory_human | sed -e 's/.*:\(.*\)M/\1/'
2.64
admin@ixre-egl-board30: redis-cli info memory | grep used_memory_human | sed -e 's/.*:\(.*\)M/\1/'
2.74
admin@ixre-egl-board30: redis-cli info memory | grep used_memory_human | sed -e 's/.*:\(.*\)M/\1/'
2.52

202205:
admin@ixre-egl-board64: redis-cli info memory | grep used_memory_human | sed -e 's/.*:\(.*\)M/\1/'
6.02
admin@ixre-egl-board64: redis-cli info memory | grep used_memory_human | sed -e 's/.*:\(.*\)M/\1/'
6.26
admin@ixre-egl-board64: redis-cli info memory | grep used_memory_human | sed -e 's/.*:\(.*\)M/\1/'
6.14

We can see that 202405 has some memory optimization for redis and it's not using as much memory as 202205. 0.2M memory usage difference could easily reach the memory usage threshold of 5% in 202405. 

Solution is to get the average redis memory usage before and after link flap. using 5 seconds interval and 5 times query and then get the average memory usage for redis. Also make the threshold to 10% from 5%. With this fix it's found that the redis memory check will not fail for 2405 after link flap. 

This commit also provide a fix for sometimes CPU utilization check failed for orchagent after link flap. The reason is in scaling setup (34k routes) orchagent takes more time to calm down.

Summary:
Fixes # (issue) https://github.com/sonic-net/sonic-mgmt/issues/15733

### Type of change

- [X] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [X] 202405

#### What is the motivation for this PR?
Fix test failures
#### How did you verify/test it?
OC tests run with the fix. Did not see the test failed.

### CODE REVIEW

**Review by arista-nwolfe — APPROVED** [review]
Change looks good to me, thanks!

File: tests/platform_tests/link_flap/test_cont_link_flap.py [hunk 1]
```diff
@@ -208,26 +207,25 @@ def test_cont_link_flap(self, request, duthosts, nbrhosts, enum_rand_one_per_hws
...
-        logging.info("Redis Memory at end: %s M", end_time_redis_memory)
+        end_time_redis_memory = get_avg_redis_mem_usage(duthost, 5, 5)
+        logging.info("Redis Memory at start: %f M", start_time_redis_memory)
+        logging.info("Redis Memory at end: %f M", end_time_redis_memory)
 
         # Calculate diff in Redis memory
-        incr_redis_memory = float(end_time_redis_memory) - float(start_time_redis_memory)
-        logging.info("Redis absolute  difference: %d", incr_redis_memory)
+        incr_redis_memory = end_time_redis_memory - start_time_redis_memory
+        logging.info("Redis absolute difference: %f", incr_redis_memory)
```
Comment (liamkearney-msft) [inline tests/platform_tests/link_flap/test_cont_link_flap.py:216]: nit: this isn't an absolute difference - "Redis memory difference: %f MB"

File: tests/platform_tests/link_flap/test_link_flap.py [hunk 2]
```diff
@@ -70,26 +70,25 @@ def test_link_flap(request, duthosts, rand_one_dut_hostname, tbinfo, fanouthosts
...
+    incr_redis_memory = end_time_redis_memory - start_time_redis_memory
+    logging.info("Redis absolute difference: %f", incr_redis_memory)
 
     # Check redis memory only if it is increased else default to pass
     if incr_redis_memory > 0.0:
-        percent_incr_redis_memory = (incr_redis_memory / float(start_time_redis_memory)) * 100
+        percent_incr_redis_memory = (incr_redis_memory / start_time_redis_memory) * 100
         logger.info("Redis Memory percentage Increase: %d", percent_incr_redis_memory)
-        incr_redis_memory_threshold = 10 if tbinfo["topo"]["type"] in ["m0", "mx"] else 5
+        incr_redis_memory_threshold = 15 if tbinfo["topo"]["type"] in ["m0", "mx"] else 10
```
Comment (liamkearney-msft) [inline tests/platform_tests/link_flap/test_link_flap.py:85]: maybe it's worth putting this threshold into a function in link_flap_utils, so we dont have the constants across two seperate files.
or even just put the whole memory threshold comparison into its own function there, eg:
`def validate_redis_memory_increase(tbinfo, start_mem, end_mem)`

File: tests/platform_tests/link_flap/test_cont_link_flap.py (hunk 1 above)
Comment (wumiaont) [reply tests/platform_tests/link_flap/test_cont_link_flap.py]: Resolved

**Review by liamkearney-msft — APPROVED** [review]

**Comment by arlakshm** [comment]
@wumiaont, can you please create PR for 202503 branch

**Comment by wumiaont** [comment]
> @wumiaont, can you please create PR for 202503 branch

This PR is already cherry-picked into 202503 by this: https://github.com/Azure/sonic-mgmt.msft/commit/ae6a0737c3d918f53f3cf2e6308899d59e779eb6

**Pull request merged**